### PR TITLE
Project task associations #55

### LINKED ISSUE #55: Project task associations.
The purpose is to provide project access to its tasks and in the opposite way - task should have access to its parent-project.

### PR DESCRIPTION
**Pull request by crappylime**
closes #55

### CODE REVIEW

File: spec/models/task_spec.rb [hunk 1]
```diff
@@ -1,16 +1,17 @@
 require 'rails_helper'
 
 RSpec.describe Task, type: :model do
-  subject(:task) do
-    Task.new(title: "Short text", 
-             description: "Long text", 
-             due_date: Date.today + 1.day )
-  end
-  
+  subject(:project) { FactoryGirl.create(:project) }
```
Comment (filip373) [inline spec/models/task_spec.rb]: I think that `project` is not your subject here, you can declare it using `let`.

File: spec/models/task_spec.rb [hunk 2]
```diff
@@ -29,6 +30,11 @@
       expect(task).not_to be_valid
     end
 
+    it "is not valid without existing project" do
+      task = FactoryGirl.build(:task)
+      task.save!(validate: false)
```
Comment (pospieszynski) [inline spec/models/task_spec.rb]: nie brakuje tu jakiegos expect?

Comment (crappylime) [inline spec/models/task_spec.rb]: It works that way ;) when I change it to `(validate: true)`, it won't pass the test

Comment (pospieszynski) [inline spec/models/task_spec.rb]: with expect method it's just more clear though;]

Comment (crappylime) [inline spec/models/task_spec.rb]: depends on the point of view, less code is more readable for me :trollface: however I can change it, no problem;)

File: app/models/project.rb [hunk 3]
```diff
@@ -4,4 +4,5 @@ class Project < ApplicationRecord
   belongs_to :user
   has_many :user_projects
   has_many :contributors, through: :user_projects, source: :user
+  has_many :tasks
```
Comment (tprochniewicz) [inline app/models/project.rb]: add empty line above

**Comment by tprochniewicz** [comment]
+1
@gotar

File: spec/models/project_spec.rb [hunk 4]
```diff
@@ -79,5 +79,10 @@
       project = FactoryGirl.create(:project_with_tasks)
       expect(project.tasks.length).to eql(3)
     end
+
+    it "destroys tasks when the project is removed" do
+      project = FactoryGirl.create(:project_with_tasks)
+      expect { project.destroy}.to change {project.tasks.count}.by(-3)
```
Comment (neonowy) [inline spec/models/project_spec.rb]: Imho you should write `change { Task.count }.by(-3)`.

File: spec/models/project_spec.rb [hunk 5]
```diff
@@ -73,4 +73,16 @@
...
+
+  context "with poperly setup association - project has_many tasks" do
+    it "has 3 tasks" do
+      project = FactoryGirl.create(:project_with_tasks)
+      expect(project.tasks.length).to eql(3)
+    end
+
+    it "destroys tasks when the project is removed" do
+      project = FactoryGirl.create(:project_with_tasks)
+      expect{ project.destroy }.to change{ project.tasks.count }.by(-3)
```
Comment (neonowy) [inline spec/models/project_spec.rb]: Imho you should write `change { Task.count }.by(-3)`.

Comment (neonowy) [inline spec/models/project_spec.rb]: Add space before `{`.

File: spec/models/task_spec.rb [hunk 6]
```diff
@@ -29,6 +30,13 @@
       expect(task).not_to be_valid
     end
 
+    it "raises validation error with try to save task without project" do
+      expect {
+        task = FactoryGirl.build(:task)
```
Comment (filip373) [inline spec/models/task_spec.rb]: You could move `task` initialization above `expect` to reduce its scope. Also, you can check for specific error to make sure it's exactly what you expect. Take a look: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/raise-error-matcher

**Comment by gotar** [comment]
rebase with master, squeeze and you can merge

**Pull request merged**